windows/win_file.ps1

```diff
@@ -50,6 +50,52 @@ $result = New-Object psobject @{
     changed = $FALSE
 }
 
+# Used to delete symlinks when powershell version is 4 or older
+$symlink_util = @"
+using System;
+using System.Runtime.InteropServices;
+
+namespace Ansible.Command
+{
+    public class DeleteSymLink
+    {
+        [DllImport("kernel32.dll", CharSet=CharSet.Unicode, SetLastError=true)]
+        public static extern bool RemoveDirectory(string lpPathName);
+    }
+}
+"@
+
+# Used to delete directories and files with logic on handling symbolic links
+Function Remove-File($file) {
+    Try {
+        If ($file.Attributes -band [System.IO.FileAttributes]::ReparsePoint) {
+            If ($PSVersionTable.PSVersion.Major -ge 5) {
+                Remove-Item -Path $file.FullName -Force
+            } Else {
+                Add-Type -TypeDefinition $symlink_util
```

Comment:
Minor perf nit- calling this in a recursive function gets kind of expensive (even though the generated type is cached, it's still parsing the code on each call). Ideally, the `Add-Type` should occur outside the loop.